torch/_subclasses/fake_tensor.py

```diff
@@ -156,3 +197,6 @@ def merge_devices(t):
         assert common_device is not None, f"Could not find common device for {func}"
 
         return common_device
+
+class FakeTensorMode(FakeTensor):
+    context = no_dispatch
```

Comment:
Can we do a modern style mode instead pretty please :)

Reply:
You didn't actually use context, AFAICT?

Reply:
What is a modern style mode ? I believe the `context` stopped recursion

Reply:
inherit from TorchDispatchMode

Reply:
will do.. mind linking me differences between two why one should do modern style over existing ?

Reply:
the main difference is you can store instance variables on the mode, since it is an actual object not a class

Reply:
nice... i probably don't need `setup_mode` then